app/src/main/java/com/humberto/tasky/agenda/presentation/AgendaViewModel.kt

```diff
@@ -60,13 +76,43 @@ class AgendaViewModel @Inject constructor(
         }
     }
 
+    private fun getAgendaForDate(selectedDate: LocalDate) {
+        viewModelScope.launch(Dispatchers.IO) {
+            _agendaState.update { state ->
+                state.copy(isLoadingAgendaItems = true)
+            }
+            localAgendaDataSource.getAgendaForDate(selectedDate)
+                .collect { agenda ->
+                    _agendaState.update { state ->
+                        state.copy(
+                            agendaItems = orderAgendaItems(agenda),
+                            isLoadingAgendaItems = false
+                        )
+                    }
+                }
+
+        }
+    }
+
+    private fun orderAgendaItems(agenda: Agenda): List<AgendaItemUi> {
+        val allItems = mutableListOf<AgendaItemUi>()
+        allItems.addAll(agenda.events.map { event -> event.toAgendaItemUi() })
+        allItems.addAll(agenda.tasks.map { task -> task.toAgendaItemUi() })
+        allItems.addAll(agenda.reminders.map { reminder -> reminder.toAgendaItemUi() })
+        return allItems.sortedBy { it.dateTime }
+    }
```

Comment:
Could it make more sense for your repository or local data source class to return a list of agenda items already ordered?